connector/connectortest/fanoutconsumer.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connectortest // import "go.opentelemetry.io/collector/connector/connectortest"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/internal/fanoutconsumer"
+)
+
+var errTooFewConsumers = errors.New("fanoutconsumer requires a mininum of 2 consumers")
+
+type TracesRouterTestOption struct {
```

Comment:
```suggestion
type tracesRouterTestOption struct {
```

Can we get away with not exporting this?

Reply:
We can. I changed it because the linter was complaining with:

> unexported-return: exported func WithNopTracesSink returns unexported type connectortest.tracesRouterTestOption, which can be annoying to use (revive)

Should I add an override for the lint instead?

Reply:
Let's leave it then. Thanks for confirming.